ui/components/package.json

```diff
@@ -24,7 +24,7 @@
     "build": "concurrently \"npm:build:*\"",
     "build:cjs": "swc ./src -d dist/cjs --config-file ../.cjs.swcrc",
     "build:esm": "swc ./src -d dist --config-file ../.swcrc",
-    "build:types": "tsc --emitDeclarationOnly --declaration --preserveWatchOutput",
+    "build:types": "tsc --project tsconfig.build.json",
```

Comment:
You [can't exclude from the CLI](https://github.com/microsoft/TypeScript/issues/46005) and we don't want to exclude typescripting these files during dev, so I needed to make a separate `tsconfig` file for the build step. Pulling all the build config into that file so it's in one place.